### PR TITLE
Convert `create-slot` to TypeScript

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

- Convert `create-slot.js` to TypeScript
- Convert `ad-sizes.js` to TypeScript (as `create-slot.js` depends on it)

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

More clearly define the types of ad slots and their sizes.

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/ad-sizes.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+type SizeKeys =
```
Comment (chrislomaxjones) [inline static/src/javascripts/projects/commercial/modules/ad-sizes.ts:1]: This type feels significant enough that it should be moved to `types.d.ts`?

File: static/src/javascripts/projects/commercial/modules/dfp/create-slots.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	| 'high-merch'
+	| 'high-merch-lucky'
+	| 'high-merch-paid'
+	| 'inline'
+	| 'mostpop'
+	| 'comments'
+	| 'top-above-nav'
+	| 'carrot'
+	| 'epic'
+	| 'mobile-sticky';
```
Comment (chrislomaxjones) [inline static/src/javascripts/projects/commercial/modules/dfp/create-slots.ts:80]: Another candidate for moving to `types.d.ts`?

File: static/src/javascripts/projects/commercial/types.d.ts [hunk 3]
```diff
@@ -91,8 +91,8 @@ type SlotVisibilityChangedEvent = {
 	slot: Slot;
 };
 
-type GuAdSize = {
+type GuAdSize = Readonly<{
 	width: number;
 	height: number;
-	toString: (_: void) => string;
-};
+	toString: () => string;
```
Comment (chrislomaxjones) [inline static/src/javascripts/projects/commercial/types.d.ts:97]: @mxdvl another candidate for template string type...

```suggestion
	toString: () => 'fluid' | `${number},${number}`;
```

**Review by mxdvl — APPROVED** [review]
I like the template string suggestion! Do it :)

**Pull request merged**